### PR TITLE
Add glossary entry for proxy

### PR DESCRIPTION
**Pull request by sftim**
I'm not sure how much this is needed; I'd assume readers already have access to Bing / Google / other web search engine.

Anyway, issue #5993 asks for a glossary entry that explains “proxy”, so here is one.
[Preview](https://deploy-preview-16293--kubernetes-io-master-staging.netlify.com/docs/reference/glossary/?all=true#term-proxy)

### CODE REVIEW

File: content/en/docs/reference/glossary/proxy.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ In computing, a proxy is a server that acts as an intermediary for a remote
+service.
+
+<!--more-->
+
+A client interacts with the proxy; the proxy copies the client's data to the
+actual server; the actual server replies to the proxy; the proxy sends the
+actual server's reply to the client.
+
+ [kube-proxy](/docs/reference/command-line-tools-reference/kube-proxy/) is a
```
Comment (steveperry-53) [inline content/en/docs/reference/glossary/proxy.md]: kube-proxy is no longer a proxy. In early versions of Kubernetes, kube-proxy was an actual proxy server. But now it's just a node component that is responsible for configuring iptables. Might have other responsibilities too; not sure.

Comment (sftim) [reply content/en/docs/reference/glossary/proxy.md]: You can still use the purely userland version of kube-proxy, right? It's just not very efficient.

**Pull request merged**